js/pdfs_strawberry.js

```diff
@@ -0,0 +1,140 @@
+(function ($, Drupal, drupalSettings, pdfjsLib, pdfjsWorker) {
+    'use strict';
+    Drupal.behaviors.format_strawberryfield_pdfjs = {
+        attach: function(context, settings) {
+            // The workerSrc property is a must!
+            // @TODO: Would love to simply push ther 'worker' property since we already have it loaded
+            // but i lack enough understanding of the API.
+            // See here https://github.com/mozilla/pdf.js/blob/master/src/display/api.js#L218
+            // Probably (have to check) passing an Object with that key instead of a string to pdfjsLib.getDocument($doc)
+            //  and worker being the global window['pdfjs-dist/build/pdf.worker'] should work
+            // basically doc.src='url' and doc.worker = window['pdfjs-dist/build/pdf.worker' ?
+            $('.strawberry-document-item[data-iiif-document]').once('attache_pdf')
+                .each(function (index, value) {
+                    var $document = $(this).data("iiif-document");
+                    var $theid = $(this).attr("id");
+                    var $initialpage = $(this).data("iiif-initialpage");
+                    pdfjsLib.GlobalWorkerOptions.workerSrc = '//cdn.jsdelivr.net/npm/pdfjs-dist@2.2.228/build/pdf.worker.min.js';
```

Comment:
Yes, too bad loading this. Saw the docs but would need to test around like you suggest.
```
The worker shall be built into a separate bundle: take the file "./node_modules/pdfjs-dist/build/pdf.worker.entry.js" or built a separate file that uses require('pdfjs-dist/build/pdf.worker'). PDFJS.workerSrc shall be set to point to this file.
```

Reply:
Its a bummer. Also, totally not possible to push. If i preload the worker, it takes (even when not around) a Port and start waiting for work. So when i push it into the workerSrc  (or smartly via property) PDF.js starts to initialize a new instance and fails! So bad and i get a warning. Anyway, this way at least its working (see how i don't load it in the `library` yml file at all because of that). Thanks for checking this. Now we need to implement search, download, and print!